### PR TITLE
Cleanup the call to RealtimeSegmentConverter from HLCDataManager

### PR DESCRIPTION
**Pull request by siddharthteotia**
After the removal of old star tree spec, the calls to constructor were not correct.

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/realtime/converter/RealtimeSegmentConverter.java [hunk 1]
```diff
@@ -82,14 +83,14 @@ public RealtimeSegmentConverter(MutableSegmentImpl realtimeSegment, String outpu
       List<String> invertedIndexColumns, List<String> noDictionaryColumns, List<String> varLengthDictionaryColumns,
       boolean nullHandlingEnabled) {
     this(realtimeSegment, outputPath, schema, tableName, tableConfig, segmentName, sortedColumn,
-        invertedIndexColumns, new ArrayList<>(), noDictionaryColumns, varLengthDictionaryColumns, nullHandlingEnabled);
+        invertedIndexColumns, Collections.emptyList(), noDictionaryColumns, varLengthDictionaryColumns, nullHandlingEnabled);
   }
 
   // Used in RealtimeSegmentConverterTest
   public RealtimeSegmentConverter(MutableSegmentImpl realtimeSegment, String outputPath, Schema schema,
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/realtime/converter/RealtimeSegmentConverter.java:90]: Let's remove the extra constructors and just keep one (easier to use).
The only usage of them is `RealtimeSegmentConverterTest.testNoVirtualColumnsInSchema()`, where you can make `RealtimeSegmentConverter.getUpdatedSchema()` static

File: pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java [hunk 2]
```diff
@@ -287,8 +287,7 @@ public void run() {
               new RealtimeSegmentConverter(_realtimeSegment, tempSegmentFolder.getAbsolutePath(), schema,
                   _tableNameWithType, tableConfig, realtimeSegmentZKMetadata.getSegmentName(), _sortedColumn,
                   HLRealtimeSegmentDataManager.this._invertedIndexColumns, _noDictionaryColumns,
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java:289]: (nit) Remove `HLRealtimeSegmentDataManager.this.`

File: pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java [hunk 3]
```diff
@@ -287,8 +287,7 @@ public void run() {
...
-                  _varLengthDictionaryColumns, null/*StarTreeIndexSpec*/,
-                  indexingConfig.isNullHandlingEnabled()); // Star tree not supported for HLC.
+                  _varLengthDictionaryColumns, indexingConfig.isNullHandlingEnabled());
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java:290]: We should support text index for HLC

Comment (siddharthteotia) [reply pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java:290]: It is there for LLC already. Will add for HLC as well

File: pinot-core/src/main/java/org/apache/pinot/core/realtime/converter/RealtimeSegmentConverter.java (hunk 1 above)
Comment (siddharthteotia) [reply pinot-core/src/main/java/org/apache/pinot/core/realtime/converter/RealtimeSegmentConverter.java]: done

File: pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java (hunk 2 above)
Comment (siddharthteotia) [reply pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java]: done

**Comment by siddharthteotia** [comment]
@Jackie-Jiang , addressed the review comments.

File: pinot-core/src/test/java/org/apache/pinot/realtime/converter/RealtimeSegmentConverterTest.java [hunk 4]
```diff
@@ -46,11 +46,7 @@ public void testNoVirtualColumnsInSchema() {
     String segmentName = "segment1";
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/realtime/converter/RealtimeSegmentConverterTest.java:46]: Remove the unused table config

File: pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java [hunk 5]
```diff
@@ -286,9 +287,8 @@ public void run() {
           RealtimeSegmentConverter converter =
               new RealtimeSegmentConverter(_realtimeSegment, tempSegmentFolder.getAbsolutePath(), schema,
                   _tableNameWithType, tableConfig, realtimeSegmentZKMetadata.getSegmentName(), _sortedColumn,
-                  HLRealtimeSegmentDataManager.this._invertedIndexColumns, _noDictionaryColumns,
-                  _varLengthDictionaryColumns, null/*StarTreeIndexSpec*/,
-                  indexingConfig.isNullHandlingEnabled()); // Star tree not supported for HLC.
+                  _invertedIndexColumns, Collections.emptyList(), _noDictionaryColumns, _varLengthDictionaryColumns,
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java:290]: Are you planning to add text index support for HLC in a separate PR?

File: pinot-core/src/test/java/org/apache/pinot/realtime/converter/RealtimeSegmentConverterTest.java (hunk 4 above)
Comment (siddharthteotia) [reply pinot-core/src/test/java/org/apache/pinot/realtime/converter/RealtimeSegmentConverterTest.java:44]: done

**Pull request merged**